### PR TITLE
Fix slow opening of settings screen

### LINKED ISSUE #1193: Settings open slowly (Pixel XL, Android 8)
On my Pixel XL running Android 8 it takes about 4 seconds until the settings show up after clicking the menu item. That's super slow and feels like the app is frozen.

### PR DESCRIPTION
**Pull request by aperfilyev**
Closes #1193

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java [hunk 1]
```diff
@@ -319,37 +329,78 @@ public CharSequence getSummary() {
...
+            final LocaleDescriptor[] descriptors = locales.toArray(new LocaleDescriptor[usableCount]);
+            Arrays.sort(descriptors, 0, usableCount);
+            return descriptors;
         }
 
-        setEntries(entries);
-        setEntryValues(values);
+        @Override
+        protected void onPostExecute(Pair<String[], String[]> pair) {
+            if (isCancelled()) return;
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java]: \[Checkstyle\] ERROR: 'if' construct must use '\{\}'s\.

File: app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java [hunk 2]
```diff
@@ -319,37 +329,78 @@ public CharSequence getSummary() {
...
+                            CharacterValidator characterValidator, Collection<String> shippingLocales) {
+            this.characterValidator = characterValidator;
+            this.shippingLocales = shippingLocales;
+            this.systemDefaultLanguage = systemDefaultLanguage;
+            this.weakListPreference = new WeakReference<>(listPreference);
+        }
 
-        this.entriesLocale = currentLocale;
+        @Override
+        protected Pair<String[], String[]> doInBackground(Void... voids) {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java]: \[PMD\] INFO: The method 'doInBackground\(Void\)' has a NCSS line count of 14\.

File: app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java [hunk 3]
```diff
@@ -319,37 +329,78 @@ public CharSequence getSummary() {
...
-        // We leave room for "System default".
-        final String[] entries = new String[count + 1];
-        final String[] values = new String[count + 1];
+        /**
+         * Not every locale we ship can be used on every device, due to
+         * font or rendering constraints.
+         * <p>
+         * This method filters down the list before generating the descriptor array.
+         */
+        private LocaleDescriptor[] getUsableLocales() {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java]: \[PMD\] INFO: The method 'getUsableLocales\(\)' has a NCSS line count of 14\.

File: app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java (hunk 1 above)
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java]: \[PMD\] INFO: Avoid using if statements without curly braces

File: app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java [hunk 4]
```diff
@@ -319,37 +329,78 @@ public CharSequence getSummary() {
...
+            this.characterValidator = characterValidator;
+            this.shippingLocales = shippingLocales;
+            this.systemDefaultLanguage = systemDefaultLanguage;
+            this.weakListPreference = new WeakReference<>(listPreference);
         }
 
-        final LocaleDescriptor[] descriptors = getUsableLocales();
-        final int count = descriptors.length;
+        @Override
+        protected Pair<String[], String[]> doInBackground(Void... voids) {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/widget/LocaleListPreference.java]: \[PMD\] INFO: The method 'doInBackground\(Void\)' has a NCSS line count of 14\.

**Comment by ekager** [comment]
Hey @aperfilyev thanks so much for your contribution! @pocmo should we suppress these warnings or will the methods need to be broken up?

**Pull request merged**